doc/development-context-aware-applications/v2/how-to-update-and-query-context-information.md

```diff
@@ -0,0 +1,243 @@
+Processes running as part of your application architecture that update
+context information using REST operations that the Context Broker GE
+exports, are said to play a **Context Producer** role.  As an example,
+let’s consider an application for reviewing restaurants (let’s call it
+NiceEating). The client part of that application running on the
+smartphone of users would play the role of Context Producer, enabling
+them to rate restaurants.
+
+On the other hand, processes running as part of your application
+architecture that query context information using REST operations that
+the Context Broker GE exports are said to play a **Context Consumer**
+role.  Continuing with our NiceEating app, the mobile application
+running on the smartphone of users and enabling them to query for the
+rating of restaurants would play the role of Context Consumer.  Note
+that a given part of your application may play both the Context Producer
+and Context Consumer roles.  This would be the case of the mobile client
+of the NiceEating app enabling end users to review and query about reviews
+of restaurants.
+
+Entities that would be relevant to the NiceEating application are of
+type Restaurant, Client and Review. 
+In this guide, the restaurant Elizalde is used as an example. In order to get its id for perform the different requests, we can use:
+
+    GET <cb_host>:<cb_port>/v2/entities?type=Restaurant&q=name==Elizalde&attrs=name&options=keyValues 
+    [
+      {
+        "id": "0115206c51f60b48b77e4c937835795c33bb953f",
+        "type": "Restaurant",
+        "name": "Elizalde"
+      }
+    ]
+
+When a given user reviews a restaurant (e.g. in a scale from 0 to 5, “Client1234” scores
+“4” for the Elizalde restaurant and describes his/her experience with "Cheap and nice place to eat.") the smartphone application plays the Context Producer role **creating** a Review entity by issuing the
+following HTTP request :
+
+    POST <cb_host>:<cb_port>/v2/entities?options=keyValues
+    {
+      "type": "Review",
+      "id": "review-Elizalde-34",
+      "author": "Client1234",
+      "itemReviewed": "0115206c51f60b48b77e4c937835795c33bb953f",
+      "reviewBody": "Cheap and nice place to eat.",
+      "ratingValue": 4
+    }
+
+Each time a new Review entity is created, the average rating for the
+corresponding restaurant is recalculated by the application backend
+which (playing also the role of Context Producer) **updates** the
+Restaurant entity accordingly:
+
+    PUT <cb_host>:<cb_port>/v2/entities/0115206c51f60b48b77e4c937835795c33bb953f/attrs/aggregateRating
```

Comment:
options=keyValues is missing here, otherwise this should not work, shouldn't it, @fgalan ?